src/clims/configuration/work_batch_definition.py

```diff
@@ -34,34 +32,28 @@ def on_button_click1xx():
     """
 
     @classmethod
-    def full_name(cls):
+    def cls_full_name(cls):
         # Corresponds to 'full_name' in serializer
         return cls.type_full_name_cls()
 
     @classmethod
     def buttons(cls):
         buttons = list()
         for _, v in iteritems(cls.__dict__):
-            if callable(v) and hasattr(v, HOOK_TAG):
+            if callable(v) and hasattr(v, HOOK_TAG) and \
+                    hasattr(v, HOOK_TYPE) and getattr(v, HOOK_TYPE) == 'button':
                 b = Button(name=v.__name__, caption=getattr(v, HOOK_TAG))
                 buttons.append(b)
         return buttons
 
-    @classmethod
-    def trigger_script(cls, event_type, event_tag, workbatch):
-        class_dict = dict(cls.__dict__)
-        for _, v in iteritems(class_dict):
-            if cls._matches_event_type(v, event_type):
-                if hasattr(v, HOOK_TAG) and getattr(v, HOOK_TAG) == event_tag:
-                    # TODO: None as self is not pretty
-                    v(None, workbatch)
-                elif not hasattr(v, HOOK_TAG):
-                    v(None, workbatch)
+    def trigger(self, event):
+        class_dict = dict(self.__class__.__dict__)
```

Comment:
Isn't `__dict__` already a dict? I.e. the `dict` function not necessary?

Reply:
lol

Reply:
Better to be safe than sorry! :D